Mage.Sets/src/mage/cards/b/BelisariusCawl.java

```diff
@@ -0,0 +1,123 @@
+package mage.cards.b;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.common.TapTargetCost;
+import mage.abilities.costs.common.TapVariableTargetCost;
+import mage.abilities.dynamicvalue.common.GetXValue;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.cards.*;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterControlledCreaturePermanent;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.predicate.permanent.TappedPredicate;
+import mage.game.Game;
+import mage.game.permanent.token.WhiteAstartesWarriorToken;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetControlledPermanent;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public final class BelisariusCawl extends CardImpl {
+
+    private static final FilterControlledPermanent artifactFilter =
+        new FilterControlledPermanent("untapped artifacts you control");
+
+    static {
+        artifactFilter.add(TappedPredicate.UNTAPPED);
+        artifactFilter.add(CardType.ARTIFACT.getPredicate());
+    }
+
+    private static final FilterControlledPermanent creatureFilter =
+        new FilterControlledCreaturePermanent("untapped creatures you control");
+
+    static {
+        creatureFilter.add(TappedPredicate.UNTAPPED);
+    }
+
+    public BelisariusCawl(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT, CardType.CREATURE}, "{2}{W}{U}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(4);
+
+        // Ultima Founding -- {T}, Tap two untapped artifacts you control: Create a 2/2 white Astartes Warrior creature token with vigilance.
+        Ability ultima = new SimpleActivatedAbility(Zone.BATTLEFIELD, new CreateTokenEffect(new WhiteAstartesWarriorToken()), new TapSourceCost());
+        ultima.addCost(new TapTargetCost(new TargetControlledPermanent(2, 2, artifactFilter, false)));
+        ultima.withFlavorWord("Ultima Founding");
+        this.addAbility(ultima);
+
+        // Master of Machines -- {T}, Tap X untapped creatures you control: Look at the top X cards of your library. You may reveal an artifact card from among them and put it into your hand. Put the rest on the bottom of your library in a random order.
+        Ability master = new SimpleActivatedAbility(Zone.BATTLEFIELD, new BelisariusCawlEffect(), new TapSourceCost());
```

Comment:
Is there a reason you can't use `LookLibraryAndPickControllerEffect` here like the other PR?

Reply:
I do not think there was an actual reason. The other PR is more succinct, should probably merge that one.